cli/server/server.go

```diff
@@ -38,43 +38,50 @@ func main() {
 	envconfig.MustProcess("jwt", &jwtConfig)
 	jwt := service.NewJWT(jwtConfig.SigningKey)
 
-	// routing
+	// cors options
+	corsOptions := cors.Options{
+		AllowedOrigins: []string{"*"},
+		AllowedMethods: []string{"GET", "POST", "PUT", "OPTIONS"},
+		AllowedHeaders: []string{"Location", "Authorization"},
+	}
+
+	// loger
+	logger := service.NewLogger()
+
 	r := chi.NewRouter()
-	r.Use(middleware.Logger)
-	r.Use(middleware.Recoverer)
-	r.Use(func(h http.Handler) http.Handler {
-		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			headers := w.Header()
-			headers.Set("Access-Control-Allow-Origin", "*")
-			headers.Set("Access-Control-Allow-Methods", "GET, POST, OPTIONS")
-			headers.Set("Access-Control-Allow-Headers", "Location, Authorization")
-			if r.Method == "OPTIONS" {
-				return
-			}
-			h.ServeHTTP(w, r)
-		})
+
+	ms := []func(http.Handler) http.Handler{
+		middlewarechi.Recoverer,
+		cors.New(corsOptions).Handler,
+		middlewarechi.RequestLogger(&middlewarechi.DefaultLogFormatter{Logger: logger}),
+	}
+
+	msProtected := append(ms, jwt.Middleware)
+
+	r.Route("/api", func(r chi.Router) {
+		r.Use(msProtected...)
+		r.Get("/me", handler.Me(userRepo))
 	})
 
 	r.Get("/login", handler.Login(oauth))
```

Comment:
Recoverer and logger middlewares must be used for all routers

Reply:
addressed by 28c54c4